modules/xml-sitemap/sitemap.php

```diff
@@ -0,0 +1,595 @@
+	}
+
+	$post_types_in = array ();
+	foreach ( $post_types as $post_type ) {
+		$post_types_in[] = $wpdb->prepare( '%s', $post_type );
+	}
+	$post_types_in_string = implode( ', ', $post_types_in );
+
+	/**
+	 * Filter limit of entries to include in news sitemap.
+	 *
+	 * @since 3.9
+	 *
+	 * @param int $count Number of entries to include in news sitemap.
+	 */
+	$limit        = apply_filters( 'jetpack_sitemap_news_sitemap_count', 1000 );
+	$cur_datetime = current_time( 'mysql', true );
+
+	$query = $wpdb->prepare( "
+		SELECT p.ID, p.post_title, p.post_type, p.post_date, p.post_name, p.post_date_gmt, GROUP_CONCAT(t.name SEPARATOR ', ') AS keywords
+		FROM
+			$wpdb->posts AS p LEFT JOIN $wpdb->term_relationships AS r ON p.ID = r.object_id
+			LEFT JOIN $wpdb->term_taxonomy AS tt ON r.term_taxonomy_id = tt.term_taxonomy_id AND tt.taxonomy = 'post_tag'
+			LEFT JOIN $wpdb->terms AS t ON tt.term_id = t.term_id
+		WHERE
+			post_status='publish' AND post_type IN ( {$post_types_in_string} ) AND post_date_gmt > (%s - INTERVAL 2 DAY)
+		GROUP BY p.ID
+		ORDER BY p.post_date_gmt DESC LIMIT %d", $cur_datetime, $limit );
+
+	header( 'Content-Type: application/xml' );
+	ob_start();
+	echo '<?xml version="1.0" encoding="UTF-8"?>' . "\n";
+	?>
+	<!-- generator="jetpack" -->
+	<urlset xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance"
+	        xsi:schemaLocation="http://www.sitemaps.org/schemas/sitemap/0.9 http://www.sitemaps.org/schemas/sitemap/0.9/sitemap.xsd"
+	        xmlns="http://www.sitemaps.org/schemas/sitemap/0.9"
+	        xmlns:news="http://www.google.com/schemas/sitemap-news/0.9"
+	        xmlns:image="http://www.google.com/schemas/sitemap-image/1.1"
+		>
+		<?php
+		$posts = $wpdb->get_results( $query );
+		foreach ( $posts as $post ):
+
+			/**
+			 * Filter condition to allow skipping specific posts in news sitemap.
+			 *
+			 * @since 3.9
+			 *
+			 * @param bool $skip Current boolean. False by default, so no post is skipped.
+			 * @param WP_POST $post Current post object.
+			 */
+			if ( apply_filters( 'jetpack_sitemap_news_skip_post', false, $post ) ) {
+				continue;
+			}
+
+			$GLOBALS['post']                       = $post;
+			$url                                   = array ();
+			$url['loc']                            = get_permalink( $post->ID );
+			$news                                  = array ();
+			$news['news:publication']['news:name'] = get_bloginfo_rss( 'name' );
+			if ( function_exists( 'get_blog_lang_code' ) ) {
+				$news['news:publication']['news:language'] = get_blog_lang_code();
+			}
+			$news['news:publication_date'] = jetpack_w3cdate_from_mysql( $post->post_date_gmt );
+			$news['news:title']            = get_the_title_rss();
+			if ( $post->keywords ) {
+				$news['news:keywords'] = html_entity_decode( ent2ncr( $post->keywords ), ENT_HTML5 );
+			}
+			$url['news:news'] = $news;
+
+			// Add image to sitemap
+			if ( current_theme_supports( 'post-thumbnails' ) && has_post_thumbnail( $post->ID ) ) {
```

Comment:
I wonder if we should use `Jetpack_Post_Images:: get_image()` here instead? It would allow us to return an image, even if the theme doesn't support Featured Images.